ocw/lib/cleanup.py

```diff
@@ -12,31 +12,30 @@
 
 
 def cleanup_run():
-    cfg = ConfigFile()
-    if cfg.has('cleanup'):
-        for vault_namespace in cfg.getList(['cleanup', 'namespaces'], cfg.getList(['vault', 'namespaces'], [''])):
+    feature = 'cleanup'
+    if ConfigFile().has(feature):
+        for namespace in PCWConfig.get_namespaces_for(feature, fallback_to_default=True):
             try:
-                providers = cfg.getList(['cleanup.namespace.{}'.format(vault_namespace), 'providers'],
-                                        ['ec2', 'azure', 'gce'])
-                logger.debug("[{}] Run cleanup for {}".format(vault_namespace, ','.join(providers)))
+                providers = PCWConfig.get_providers_for(feature, namespace)
+                logger.debug("[{}] Run cleanup for {}".format(namespace, ','.join(providers)))
                 if 'azure' in providers:
-                    Azure(vault_namespace).cleanup_all()
+                    Azure(namespace).cleanup_all()
 
                 if 'ec2' in providers:
-                    EC2(vault_namespace).cleanup_all()
+                    EC2(namespace).cleanup_all()
 
                 if 'gce' in providers:
-                    GCE(vault_namespace).cleanup_all()
+                    GCE(namespace).cleanup_all()
 
             except Exception as e:
-                logger.exception("[{}] Cleanup failed!".format(vault_namespace))
-                send_mail('{} on Cleanup in [{}]'.format(type(e).__name__, vault_namespace), traceback.format_exc())
+                logger.exception("[{}] Cleanup failed!".format(namespace))
+                send_mail('{} on Cleanup in [{}]'.format(type(e).__name__, namespace), traceback.format_exc())
 
 
 def list_clusters():
-    cfg = ConfigFile()
-    if cfg.has('clusters'):
-        for vault_namespace in cfg.getList(['clusters', 'namespaces'], ['']):
+    feature = 'clusters'
```

Comment:
I don't see the reason to use a variable here. I think it makes it even less readable.

Reply:
ok